crates/nao/src/lib.rs

```diff
@@ -333,3 +359,33 @@ impl Display for Network {
         }
     }
 }
+
+fn extract_version_num(input:&str) -> Option<String>{
```

Comment:
Please don't abbreviate!

```suggestion
fn extract_version_number(input:&str) -> Option<String>{
```